keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1004 @@
+  CRD](https://cloud.google.com/traffic-director/docs/security-proxyless-setup#:~:text=Save%20the%20following%20TrustConfig%20YAML%20configuration%20to%20tell%20your%20cluster%20how%20to%20trust%20the%20issued%20certificates%3A)
+  to distribute trust anchors to workloads.
+
+Each of these approaches has downsides:
+
+* Creating a ConfigMap in every namespace requires granting over-broad
+  permissions to the responsible controller (create ConfigMaps in **all**
+  namespaces).
+
+* Static trust anchor sets embedded directly in configuration complicate the job
+  of rotating the backing CA's roots of trust.
+
+* CRDs work well, but are inappropriate for use by core APIs.
+
+Each of these approaches would be simplified by an in-tree mechanism for
+distributing trust anchor sets.
+
+### Goals
+
+* Provide a Kubernetes data type that holds X.509 trust anchors for use wherever
+  a set of X.509 trust anchors is needed.
+
+* Make it easy for certificates/v1 signer implementations to publish their
+  relevant trust anchors using a TrustAnchorSets object.
+
+* Make it easy for workloads that need access to the trust anchors that back a
+  particular certificates/v1 signer to access them within the container
+  filesystem.
+
+* Allow cluster operators to configure webhook and aggregrated API server
+  backends using that use TrustAnchorSets instead of inline PEM data.
+
+### Non-Goals
+
+* Distribute a default set of "system" trust anchors using TrustAnchorSets.
+  (This is a natural extension, though).
+
+* Handle trust anchors expressed in other forms than PEM-wrapped, DER-formatted
+  X.509.
+
+## Proposal
+
+This proposal is centered around a new cluster-scoped TrustAnchorSets resource,
+initially in the certificates/v1alpha1 API group.  The TrustAnchorSets object
+can be thought of as a specialized configmap tailored to the X.509 trust anchor
+use case.  Introducing a dedicated type allows us to attach different RBAC
+policies to TrustAnchorSets, which will typically be wide-open for reading, but
+locked-down for writing.
+
+Each TrustAnchorSets object is a container for multiple named trust anchor sets.
+Most use cases will simply using a single trust anchor set named default, but
+mutual TLS use cases involving multiple federating meshes will require the
+ability to express multiple, non-commingled trust anchor sets.
+
+Like CertificateSigningRequest, TrustAnchorSets are scoped to a particular
+signer using a `signerName` field, which will also be usable in RBAC
+ClusterRoleBindings.  It is possible to grant specific users the ability
+read and/or update all TrustAnchorSets that correspond to a specific signer.
+
+To enable consumption by workloads, the new `trustAnchorSets` Kubelet projected
+volume source allows projecting a trust anchor set from a particular signer into
+the container filesystem, with the contents of the projected files updating as
+the corresponding TrustAnchorSets are updated.
```

Comment:
Does the kubelet validate if the pod SA can read said trust anchor set?

Reply:
Based on the conversation SIG Auth, no it doesn't.  I've clarified that this will effectively allow all pods in the cluster to *read* any TrustAnchorSets object.

Reply:
Pods don't have to be able to read configmaps that are referenced in their podspec. I'd expect this to work essentially the same way.